ext/pcre/php_pcre.c

```diff
@@ -638,10 +645,19 @@ PHPAPI pcre_cache_entry* pcre_get_compiled_regex_cache_ex(zend_string *regex, in
 	   back the compiled pattern, otherwise go on and compile it. */
 	zv = zend_hash_find(&PCRE_G(pcre_cache), key);
 	if (zv) {
-		if (key != regex) {
-			zend_string_release_ex(key, 0);
+		pcre_cache_entry *pce = (pcre_cache_entry*)Z_PTR_P(zv);
+#ifdef HAVE_PCRE_JIT_SUPPORT
```

Comment:
This is hard to read. How about:

```
#ifdef HAVE_PCRE_JIT_SUPPORT
bool recompile = (bool)(pce->preg_options & PREG_JIT_ATTEMPTED) != jit_enabled;
#else
bool recompile = false;
#endif
if (recompile) {
    ...
}
```
The optimizer will remove the redundant if.

Reply:
done